hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -127,16 +133,35 @@ public Writer appendBlock(HoodieLogBlock block)
     return rolloverIfNeeded();
   }
 
+  private boolean isLogFileSealed(Path path) throws IOException {
+    long currentOffset = this.output.getPos();
+    FSDataInputStream input = null;
+    try {
+      input = fs.open(path);
+      input.readFully(currentOffset - magicBuffer.length, magicBuffer);
+      if (Arrays.equals(magicBuffer, HoodieLogFormat.END_MARKER)) {
+        return true;
+      }
+      return false;
+    } finally {
+      input.close();
+    }
+  }
+
   private Writer rolloverIfNeeded() throws IOException, InterruptedException {
     // Roll over if the size is past the threshold
     if (getCurrentSize() > sizeThreshold) {
-      //TODO - make an end marker which seals the old log file (no more appends possible to that file).
       log.info("CurrentSize " + getCurrentSize() + " has reached threshold " + sizeThreshold
           + ". Rolling over to the next version");
       HoodieLogFile newLogFile = logFile.rollOver(fs);
+      // Create the new writer
+      Writer newWriter = new HoodieLogFormatWriter(fs, newLogFile, bufferSize, replication, sizeThreshold);
+      // End marker which seals the old log file (no more appends possible to this file)
+      // TODO : Auto-correct old log files in case failed to write END_MARKER
```

Comment:
Caveat : Failures can happen here after creating new logFile but before END_MARKER is appended to previous logfile.

Reply:
Yes, when you open a LogFile for writing - make sure that the previous log file is locked.

Reply:
I will handle locking of log files when attempting async compaction. Just putting the TODO for now.

Reply:
Why not write the end marker first? That seems like the best thing to do here, no?  then you are left with just handling the case of a partial end marker, right? 

Please file issues for known gaps like this.